src/daft-logical-plan/src/ops/set_operations.rs

```diff
@@ -5,11 +5,20 @@ use daft_core::{count_mode::CountMode, join::JoinType, utils::supertype::get_sup
 use daft_dsl::{lit, null_lit, resolved_col, ExprRef};
 use daft_functions::list::{explode, list_fill};
 use daft_schema::{dtype::DataType, field::Field, schema::SchemaRef};
+use indexmap::IndexSet;
 use snafu::ResultExt;
 
 use super::{Aggregate, Concat, Distinct, Filter, Project};
 use crate::{logical_plan, logical_plan::CreationSnafu, LogicalPlan};
 
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
+pub enum SetQuantifier {
+    All,
+    Distinct,
+    AllByName,
+    DistinctByName,
+}
```

Comment:
Let's keep the SetQuantifier separate from the union compatibility strategy ("positions", "names")

Something like,

```rust
pub enum SetQuantifier {
    All,
    Distinct,
}

pub enum SetCollation {
    Positions,
    Names,
}
```

Reply:
We will also need an enum/flag for outer unions

Reply:
I'm not convinced we need separate ones here. My implementation was inspired by sqlparser-rs's [SetQuantifier](https://github.com/apache/datafusion-sqlparser-rs/blob/affe8b549884a351ead4f35aa8bdf4cae8c93e4b/src/ast/query.rs#L225)

Reply:
I will add split out the quantifier & strategy, but won't add outer union at this time.